lms/djangoapps/completion_api/views.py

```diff
@@ -254,9 +259,21 @@ def get(self, request):
         Handler for GET requests.
         """
         self.paginator = self.pagination_class()  # pylint: disable=attribute-defined-outside-init
+        mobile_only = (self.request.query_params.get('mobile_only', 'false')).lower() == 'true'
 
         # Paginate the list of active enrollments, annotated (manually) with a student progress object.
-        enrollments = CourseEnrollment.objects.filter(user=self.get_user(), is_active=True).order_by('course_id')
+        enrollments = CourseEnrollment.objects.filter(
+            user=self.get_user(),
+            is_active=True).order_by('course_id')
+
+        if mobile_only:
+            course_keys = []
+            for course_enrollment in enrollments:
+                course_keys.append(course_enrollment.course_id)
+                course_overview_list = CourseOverview.objects.filter(id__in=course_keys, mobile_available=True)
```

Comment:
This line needs to be bumped out too.  It's the one that builds the query from the course_keys list you're trying to build with the for-loop.

Reply:
By god you're right!